### PR TITLE
Update network tests to use random ports

### PR DESCRIPTION
**Pull request by halfalicious**
The network tests can sometimes interfere with each other - update them
to use random ports which should reduce the chances of test failures due
to test interference.

### CODE REVIEW

File: libp2p/Network.h [hunk 1]
```diff
@@ -37,7 +37,8 @@ namespace dev
 namespace p2p
 {
 
-static const unsigned short c_defaultListenPort = 30303;
+static constexpr unsigned short c_defaultListenPort = 30303;
+static constexpr const char* c_localhostIp = "127.0.0.1";
```
Comment (chfast) [inline libp2p/Network.h]: The `static` is redundant. `constexpr` makes the linkage static.
If this is only used in tests, maybe define it only there?

**Comment by halfalicious** [comment]
Run failed because it exceeded the max CircleCI runtime, is there a way we can increase it or should we look into optimizing the tests?

**Comment by NAKsir-melody** [comment]
Any duplication check for the generated random port numbers?

**Comment by chfast** [comment]
> Any duplication check for the generated random port numbers?

It will not be effective because they are going to run in multiple processes.

**Comment by gumb0** [comment]
It looks like some tests are now hanging sometimes in the endless loop when trying to find port number

File: test/unittests/libp2p/net.cpp [hunk 2]
```diff
@@ -289,7 +297,7 @@ class TestUDPSocketHost : UDPSocketEvents, public TestHost
             }
             catch (std::exception const&)
             {
-                ++port;
+                getRandomPortNumber();
```
Comment (gumb0) [inline test/unittests/libp2p/net.cpp]: You forgot to assign it to `port` here

**Pull request merged**